lib/salus/scanners/slither.rb

```diff
@@ -0,0 +1,94 @@
+require 'json'
+require 'salus/scanners/base'
+
+# Slither (https://github.com/crytic/slither) is a Python-based static
+# analysis tool for Solidity smart contracts
+
+module Salus::Scanners
+  class Slither < Base
+    include Salus::Formatting
+
+    def should_run?
+      # only support truffle/hardhat for now
+      return false if !@repository.package_json_present? || !has_package_config
+
+      @repository.sol_file_present? != false
+    end
+
+    def run
+      # Most Solidity projects will be Hardhat (https://hardhat.org/) or Truffle (https://trufflesuite.com/),
+      # which both use NPM to manage Solidity dependencies. We should NPM install in these cases
+      # to ensure Slither has the needed dependencies for scanning
+
+      pre_slither_commands = ['npm install', 'npm config set user 0']
+      pre_slither_commands.each do |pcmd|
+        shell_return = run_shell(pcmd, chdir: @repository.path_to_repo)
+        if !shell_return.success?
+          err_msg = pcmd + ' failed! ' + shell_return.stderr
+          report_error(err_msg)
+          report_stderr(err_msg)
+          return report_failure
+        end
+      end
+
+      slither_cmd = 'slither . --json - --exclude-informational --exclude-optimization'
```

Comment:
Should we add a TODO comment to update this when we support configurability?

Reply:
added comment.